### PR TITLE
[world rebuild][vcpkg scripts] Test, fix, extend vcpkg_execute_required_process

### PR DESCRIPTION
**Pull request by dg0yt**
Fixes `OUTPUT_VARIABLE`, `ERROR_VARIABLE`. 

vcpkg always wants logfiles. CMake (`execute_process`) doesn't send stdout/stderr to files and variables at the same time, and even "the precedence is not specified". So simultaneous output has to be handled in `vcpkg_execute_required_process`.

At this occasion, port the arguments for stripping trailing whitespace.

`tensorflow-common` implicitly relied on the broken behavior. I chose the minimal fix: drop the pointless calls, just set the empty value.

### CODE REVIEW

**Comment by dg0yt** [comment]
`tensorflow`, `tensorflow-cc`: 
~~~
WARNING: Download from https://storage.googleapis.com/mirror.tensorflow.org/github.com/tensorflow/runtime/archive/6ca793b5d862ef6c50f242d77a811f06cce9b60a.tar.gz failed: class java.io.FileNotFoundException GET returned 404 Not Found
...
WARNING: Download from https://storage.googleapis.com/mirror.tensorflow.org/github.com/llvm/llvm-project/archive/0538e5431afdb1fa05bdcedf70ee502ccfcd112a.tar.gz failed: class java.io.FileNotFoundException GET returned 404 Not Found
...
ERROR: An error occurred during the fetch of repository 'local_config_python':
   Traceback (most recent call last):
	File "/mnt/vcpkg-ci/b/tensorflow/x64-linux-dbg/third_party/py/python_configure.bzl", line 271, column 40, in _python_autoconf_impl
		_create_local_python_repository(repository_ctx)
	File "/mnt/vcpkg-ci/b/tensorflow/x64-linux-dbg/third_party/py/python_configure.bzl", line 214, column 22, in _create_local_python_repository
		_check_python_lib(repository_ctx, python_lib)
	File "/mnt/vcpkg-ci/b/tensorflow/x64-linux-dbg/third_party/py/python_configure.bzl", line 138, column 25, in _check_python_lib
		auto_config_fail("Invalid python library path: %s" % python_lib)
	File "/mnt/vcpkg-ci/b/tensorflow/x64-linux-dbg/third_party/remote_config/common.bzl", line 12, column 9, in auto_config_fail
		fail("%sConfiguration Error:%s %s\n" % (red, no_color, msg))
Error in fail: Configuration Error: Invalid python library path: /home/cloudtest/.local/lib/python3.10/site-packages
ERROR: /mnt/vcpkg-ci/b/tensorflow/x64-linux-dbg/WORKSPACE:15:14: fetching python_configure rule //external:local_config_python: Traceback (most recent call last):
	File "/mnt/vcpkg-ci/b/tensorflow/x64-linux-dbg/third_party/py/python_configure.bzl", line 271, column 40, in _python_autoconf_impl
		_create_local_python_repository(repository_ctx)
	File "/mnt/vcpkg-ci/b/tensorflow/x64-linux-dbg/third_party/py/python_configure.bzl", line 214, column 22, in _create_local_python_repository
		_check_python_lib(repository_ctx, python_lib)
	File "/mnt/vcpkg-ci/b/tensorflow/x64-linux-dbg/third_party/py/python_configure.bzl", line 138, column 25, in _check_python_lib
		auto_config_fail("Invalid python library path: %s" % python_lib)
	File "/mnt/vcpkg-ci/b/tensorflow/x64-linux-dbg/third_party/remote_config/common.bzl", line 12, column 9, in auto_config_fail
		fail("%sConfiguration Error:%s %s\n" % (red, no_color, msg))
Error in fail: Configuration Error: Invalid python library path: /home/cloudtest/.local/lib/python3.10/site-packages
~~~

Maybe from `tensorflow-common`:
~~~cmake
	vcpkg_execute_required_process(COMMAND ${PYTHON3} -c "import site; print(site.getusersitepackages())" WORKING_DIRECTORY ${CURRENT_BUILDTREES_DIR} LOGNAME prerequisites-pypath-${TARGET_TRIPLET} OUTPUT_VARIABLE PYTHON_LIB_PATH)
~~~
now actually setting `PYTHON_LIB_PATH`.

File: ports/tensorflow-common/tensorflow-common.cmake [hunk 1]
```diff
@@ -52,14 +53,16 @@ else()
 			if (NOT EXISTS ${CURRENT_BUILDTREES_DIR}/miniforge3)
 				vcpkg_execute_required_process(COMMAND curl -fsSLo Miniforge3.sh "https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-MacOSX-arm64.sh" WORKING_DIRECTORY ${CURRENT_BUILDTREES_DIR} LOGNAME prerequisites-miniforge3-${TARGET_TRIPLET})
```
Comment (Neumann-A) [inline ports/tensorflow-common/tensorflow-common.cmake:54]: this skips asset caching doesn't it ?

Comment (dg0yt) [reply ports/tensorflow-common/tensorflow-common.cmake:54]: (And if Microsoft would care for fixing binary caching for large artifacts, they might save a lot of CI and CO2.)

**Comment by jimwang118** [comment]
execute-required-process feature passed with following triplet:

```
x64-windows
```

File: ports/tensorflow-common/tensorflow-common.cmake [hunk 2]
```diff
@@ -72,11 +71,9 @@ else()
 	else()
 		vcpkg_execute_required_process(COMMAND ${PYTHON3} -m pip install -U pip numpy WORKING_DIRECTORY ${CURRENT_BUILDTREES_DIR} LOGNAME prerequisites-pip-${TARGET_TRIPLET})
 	endif()
-
-	vcpkg_execute_required_process(COMMAND ${PYTHON3} -c "import site; print(site.getusersitepackages())" WORKING_DIRECTORY ${CURRENT_BUILDTREES_DIR} LOGNAME prerequisites-pypath-${TARGET_TRIPLET} OUTPUT_VARIABLE PYTHON_LIB_PATH)
 endif()
 set(ENV{PYTHON_BIN_PATH} "${PYTHON3}")
-set(ENV{PYTHON_LIB_PATH} "${PYTHON_LIB_PATH}")
+set(ENV{PYTHON_LIB_PATH} "")
```
Comment (BillyONeal) [inline ports/tensorflow-common/tensorflow-common.cmake:76]: It's probably worth adding a comment as to why this is here but not if it's the only change.

File: scripts/cmake/vcpkg_execute_required_process.cmake [hunk 3]
```diff
@@ -1,6 +1,6 @@
 function(vcpkg_execute_required_process)
     cmake_parse_arguments(PARSE_ARGV 0 arg
-        "ALLOW_IN_DOWNLOAD_MODE"
+        "ALLOW_IN_DOWNLOAD_MODE;OUTPUT_STRIP_TRAILING_WHITESPACE;ERROR_STRIP_TRAILING_WHITESPACE"
```
Comment (BillyONeal) [inline scripts/cmake/vcpkg_execute_required_process.cmake:3]: Need docs: https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/maintainers/functions/vcpkg_execute_required_process.md

**Review by BillyONeal — COMMENTED** [review]
Thanks for adding a test and fixing the problem!

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
Tagging team review because it's technically a backwards incompatible change to scripts but I think we should take it anyway.

**Comment by BillyONeal** [comment]
@AugP
@ras0219-msft
@data-queue
@JavierMatosD
and I discussed this today.

@AugP This means it would break people who upgrade vcpkg but don't update their baseline?

@ras0219-msft Yes that is the concern. But we didn't document this as being broken like that. It's borderline but I think it's OK.

@data-queue Could it be made into a script port?

@BillyONeal I think execute_required_process is too 'fundamental' to do that.

@AugP will mention this in the blog post for July as a known breaking change.

I'll push the merge button just before Friday's build at 9pm pacific.

File: scripts/cmake/vcpkg_execute_required_process.cmake (hunk 3 above)
Comment (BillyONeal) [reply scripts/cmake/vcpkg_execute_required_process.cmake:3]: https://github.com/microsoft/vcpkg-docs/pull/348

**Comment by BillyONeal** [comment]
Thanks for the fix!

**Pull request merged**